persist/sqldb/workflow_archive.go

```diff
@@ -70,9 +70,15 @@ func (r *workflowArchive) ArchiveWorkflow(wf *wfv1.Workflow) error {
 		return err
 	}
 	return r.session.Tx(context.Background(), func(sess sqlbuilder.Tx) error {
-		// We assume that we're much more likely to be inserting rows that updating them, so we try and insert,
-		// and if that fails, then we update.
-		// There is no check for race condition here, last writer wins.
+		_, err := sess.
```

Comment:
we take a different, alternative approach - delete any records from the archive table (which will cascade to labels) and then just insert the archive record and labels

Reply:
👍